server.go

```diff
@@ -177,3 +181,20 @@ func (s *Server) DeserializeKE2(ke2 []byte) (*message.KE2, error) {
 func (s *Server) DeserializeKE3(ke3 []byte) (*message.KE3, error) {
 	return s.Parameters.DeserializeKE3(ke3)
 }
+
+// DeserializeAKEState sets the interal state of the AKE server from the given
+// bytes
+func (s *Server) DeserializeAKEState(state []byte) error {
+	l := s.Parameters.MAC.Size()
+	if len(state) != 2*l {
```

Comment:
I re-read the draft IETF spec just to make sure that the MAC and the session key are always the same length.

Reply:
Not exactly: the MAC length is set to Nm, and the session secret length is Nh. Both depend on the chosen algorithm, which is often the same but can be different.